lib/src/models/wallet.dart

```diff
@@ -81,7 +86,28 @@ class Wallet {
       }
     });
 
-    return Wallet(keyPair: Jwk.fromJson(jwk).toKeyPair());
+    var privK = Jwk.fromJson(jwk).toKeyPair() as RsaKeyPairData;
+
+    if (privK.dp == null ||
```

Comment:
it would be good to test them individually and generate them individually as well in the case of a jwk that contains some of the optional properties

Reply:
I think the one concern I have is that the spec says if one of these fields is provided, all three should be provided. So my assumption is that if only one or two of the three is provided, it's an invalid key, and we should make it valid by generating all three for consistency.